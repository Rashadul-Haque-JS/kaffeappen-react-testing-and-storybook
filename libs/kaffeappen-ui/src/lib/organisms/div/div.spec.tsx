import { render } from '@testing-library/react';
import DivComponent from './div';
import FormComponent from '../../molecules/form/form';
import ButtonComponent from '../../atoms/button/button';
import InputComponent from '../../atoms/input/input';
import HeadlineComponent from '../../atoms/headline/headline';

describe('Div', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <DivComponent>
        <FormComponent>
          <InputComponent placeholder="användernamn" type={'text'} />
          <InputComponent placeholder="lösenord" type={'password'} />
          <ButtonComponent
            children="Logga In"
            variant="primary"
            marginTop={66}
          />
        </FormComponent>
        <ButtonComponent
          children="Registrera"
          variant="secondary"
          marginTop={2}
        />
      </DivComponent>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot successfully', () => {
    const { baseElement } = render(
      <DivComponent>
        <HeadlineComponent>Kaffeappen</HeadlineComponent>
        <FormComponent>
          <InputComponent placeholder="användernamn" type={'text'} />
          <InputComponent placeholder="lösenord" type={'password'} />
          <ButtonComponent
            children="Logga In"
            variant="primary"
            marginTop={66}
          />
        </FormComponent>
        <ButtonComponent
          children="Registrera"
          variant="secondary"
          marginTop={2}
        />
      </DivComponent>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
