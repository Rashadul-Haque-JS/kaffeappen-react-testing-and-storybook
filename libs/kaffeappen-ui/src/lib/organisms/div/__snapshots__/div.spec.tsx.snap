// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Div should match snapshot successfully 1`] = `
<body>
  <div>
    <div
      class="div__StyledDiv-sc-5071l8-0 jpgfem"
    >
      <h1
        class="headline__StyledHeadline-mv29hz-0 imWFxt"
      >
        Kaffeappen
      </h1>
      <form
        class="form__StyledForm-qsc0mr-0 eqcWSg"
      >
        <input
          class="input__StyledInput-sc-1rotb0l-0 HWiIs"
          placeholder="användernamn"
          required=""
          type="text"
        />
        <input
          class="input__StyledInput-sc-1rotb0l-0 HWiIs"
          placeholder="lösenord"
          required=""
          type="password"
        />
        <button
          class="button__StyledButton-sc-1l6wl6y-0 jakITz"
        >
          Logga In
        </button>
      </form>
      <button
        class="button__StyledButton-sc-1l6wl6y-0 gBIovO"
      >
        Registrera
      </button>
    </div>
  </div>
</body>
`;
